lisa/_assets/kmodules/lisa/fs.c

```diff
@@ -0,0 +1,755 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+
+#include <linux/fs.h>
+#include <linux/fs_context.h>
+#include <linux/seq_file.h>
+#include <linux/slab.h>
+#include <linux/user_namespace.h>
+#include <linux/percpu-rwsem.h>
+
+#include "features.h"
+#include "configs.h"
+
+static int lisa_fs_create_files(struct dentry *dentry, bool is_top_level, struct lisa_cfg *cfg);
+static struct dentry *
+lisa_fs_create_single(struct dentry *parent, const char *name,
+		      const struct inode_operations *i_ops,
+		      const struct file_operations *f_ops, umode_t mode,
+		      void *data);
+
+#define LISA_FS_SUPER_MAGIC 0xcdb11bc9
+
+/* Protect the interface. */
+static struct mutex interface_lock;
+
+int feature_param_lisa_validate(struct feature_param_entry_value *val)
```

Comment:
Why not placing it in feature_param.[c/h] ? Seems bit misplaced 
Same re: lisa_features_param

Reply:
Yes right, done